bitmapcontainer.go

```diff
@@ -893,8 +893,28 @@ func (bc *bitmapContainer) iandNotArray(ac *arrayContainer) container {
 }
 
 func (bc *bitmapContainer) iandNotRun16(rc *runContainer16) container {
-	rcb := rc.toBitmapContainer()
-	return bc.iandNotBitmapSurely(rcb)
+	if rc.isEmpty() || bc.isEmpty() {
+		// Nothing to do.
+		return bc
+	}
+
+	wordRangeStart := rc.iv[0].start / 64
+	wordRangeEnd := (rc.iv[len(rc.iv)-1].last()) / 64 // inclusive
+
+	cardinalityChange := popcntSlice(bc.bitmap[wordRangeStart : wordRangeEnd+1]) // before cardinality - after cardinality (for word range)
```

Comment:
I do not understand this part. I don't understand how you can update the cardinality with a single popcntSlice.

I suspect it is not correct... I am troubled by the fact that our test pass... but my suspicions is that the code is not correct. Can you convince me?

Reply:
Oh. Wait. I see what you are doing. Ok. It looks good.